pytensor/link/jax/dispatch/einsum.py

```diff
@@ -0,0 +1,15 @@
+import jax.numpy as jnp
+
+from pytensor.link.jax.dispatch import jax_funcify
+from pytensor.tensor.einsum import Einsum
+
+
+@jax_funcify.register(Einsum)
+def jax_funcify_Einsum(op, **kwargs):
+    subscripts = op.subscripts
+    optimize = op.optimize
```

Comment:
This was meant to be in the past tense. It's to indicate whether the inner graph is already optimized or not. If it's optimized we can inline, if not we can try again if we discover static shapes for the inputs.

Reply:
Can we use a more distinct name for that concept then? Having one thing argument "optimize" and another called "optimized" within the same object, which changes based on the context, is pretty horrible.

Reply:
I didn't want it to be verbose because it shows up in the dprint

Reply:
Also we not really giving an option not to optimize

Reply:
That's all fair, if you want `optimized` I think it's fine, provided we document why it's different to the `optimize` arg in `np.einsum` and `jnp.einsum`. 

I will point out that I made the change because tests were failing, because it was only half changed -- `optimize` was used in some places and `optimized` in others. I really think it's semantically too similar and asking for confusion.

Reply:
I reverted and added some notes. We still have to decide whether we want to allow `optimize` kwarg. It's not hard to do it. In that case we need two kwargs, one for the desired optimization, and another for whether we have managed to optimize the graph. If the user does not want optimization or if we have optimized the graph we could inline the Einsum to be further optimized with the rest of the graph?